flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/EmbeddedRocksDBStateBackend.java

```diff
@@ -305,6 +311,11 @@ private EmbeddedRocksDBStateBackend(
                 overlapFractionThreshold >= 0 && this.overlapFractionThreshold <= 1,
                 "Overlap fraction threshold of restoring should be between 0 and 1");
 
+        useIngestDbRestoreMode =
+                original.useIngestDbRestoreMode == UNDEFINED_USE_INGEST_DB_RESTORE_MODE
+                        ? config.get(USE_INGEST_DB_RESTORE_MODE)
+                        : original.useIngestDbRestoreMode;
```

Comment:
Won't this ignore `original.useIngestDbRestoreMode` if it's set to `false`?
(I guess this is a copy-paste from other fields, but they aren't boolean)

Reply:
Yep, I think we better use `Boolean`and `null` for undefined.